### PR TITLE
PWN- 8652 - Add Striga second step fragment, and refactor first step

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8652

### CODE REVIEW

File: .idea/codeStyles/Project.xml
Comment (gslevinkov) [inline .idea/codeStyles/Project.xml:1]: god damn, revert this

File: app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepContract.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.p2p.wallet.striga.ui.firststep
+
+import org.p2p.wallet.common.mvp.MvpPresenter
+import org.p2p.wallet.common.mvp.MvpView
+import org.p2p.wallet.striga.model.StrigaSignupDataType
+
+interface StrigaSignUpFirstStepContract {
+    interface View : MvpView {
+        fun updateText(newValue: String, type: StrigaSignupDataType)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepContract.kt:9]: not update text, but updateSignupField

File: app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepContract.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    }
+    interface Presenter : MvpPresenter<View> {
+        fun onTextChanged(newValue: String, type: StrigaSignupDataType)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepContract.kt:12]: onFieldChanged

we have registration fields that are changed, not texts

File: app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepFragment.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            uiKitToolbar.setOnMenuItemClickListener {
+                if (it.itemId == R.id.helpItem) {
+                    view.hideKeyboard()
+                    IntercomService.showMessenger()
+                    return@setOnMenuItemClickListener true
+                }
+                false
+            }
+
+            StrigaSignupDataType.values().forEach { dataType ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepFragment.kt:46]: cachedValues

File: app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepFragment.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    override fun updateText(newValue: String, type: StrigaSignupDataType) {
+        val view = editTextFieldsMap[type]
+        view?.setText(newValue)
+    }
+
+    private fun createEditTextsMap(): Map<StrigaSignupDataType, UiKitEditText> {
+        return with(binding) {
+            mutableMapOf<StrigaSignupDataType, UiKitEditText>().apply {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/firststep/StrigaSignUpFirstStepFragment.kt:65]: no need for apply if you will build it using buildMap

```
buildMap { 
    this[StrigaSignupDataType.EMAIL] = editTextEmail
...
}

File: app/src/main/java/org/p2p/wallet/striga/ui/secondstep/StrigaSignUpSecondStepFragment.kt [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import org.p2p.uikit.components.UiKitEditText
+import org.p2p.wallet.R
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentStrigaSignUpSecondStepBinding
+import org.p2p.wallet.intercom.IntercomService
+import org.p2p.wallet.striga.model.StrigaSignupDataType
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+typealias IV = StrigaSignUpSecondStepContract.View
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/striga/ui/secondstep/StrigaSignUpSecondStepFragment.kt:16]: typealias View 
typealias Presenter
would be enough imho

File: app/src/main/res/layout/fragment_striga_sign_up_first_step.xml [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/bg_smoke"
+    android:fillViewport="true">
+
+    <org.p2p.uikit.organisms.UiKitToolbar
+        android:id="@+id/uiKitToolbar"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_striga_sign_up_first_step.xml:11]: just toolbar is ok

File: app/src/main/res/layout/fragment_striga_sign_up_first_step.xml [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+        app:layout_constraintTop_toBottomOf="@id/uiKitToolbar">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical"
+            android:paddingBottom="48dp">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:id="@+id/viewInfo"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_striga_sign_up_first_step.xml:39]: containerInfo

File: app/src/main/res/layout/fragment_striga_sign_up_first_step.xml [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginHorizontal="16dp"
+                android:layout_marginTop="12dp"
+                android:textAppearance="@style/UiKit.TextAppearance.Regular.Title2"
+                app:hintText="@string/striga_enter_hint"
+                app:labelText="@string/striga_first_name" />
+
+            <org.p2p.uikit.components.UiKitEditText
+                android:id="@+id/editTextSurname"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_striga_sign_up_first_step.xml:124]: editTextLastName

File: app/src/main/res/layout/fragment_striga_sign_up_second_step.xml [hunk 9]
```diff
@@ -0,0 +1,222 @@
+<?xml version="1.0" encoding="utf-8"?>
+
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_striga_sign_up_second_step.xml:2]:
```suggestion
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitEditText.kt [hunk 10]
```diff
@@ -82,8 +99,17 @@ class UiKitEditText @JvmOverloads constructor(
     val length: Int
         get() = binding.editTextField.length()
 
-    fun addOnTextChangedListener(block: (String) -> Unit) {
-        binding.editTextField.doAfterTextChanged { block(it.toString()) }
+    fun addOnTextChangedListener(block: (Editable) -> Unit) {
+        val tag = viewTag ?: return
+        editTextWatcher = object : SimpleTextWatcher(tag) {
+
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitEditText.kt:105]:
```suggestion
```

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/SimpleTextWatcher.kt [hunk 11]
```diff
@@ -1,13 +1,17 @@
-package org.p2p.wallet.common.ui.textwatcher
+package org.p2p.uikit.utils
 
 import android.text.Editable
 import android.text.TextWatcher
 
-open class SimpleTextWatcher : TextWatcher {
+abstract class SimpleTextWatcher(val tag: Any) : TextWatcher {
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/SimpleTextWatcher.kt:6]: SimpleTagTextWatcher

**Review by gslevinkov — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

File: app/src/main/res/layout/fragment_striga_sign_up_first_step.xml (hunk 6 above)
Comment (Davran331122) [reply app/src/main/res/layout/fragment_striga_sign_up_first_step.xml:11]: uikittoolbar is Ok too x D

**Pull request merged**